client/python/datajunction/client.py

```diff
@@ -215,7 +216,281 @@ def data(  # pylint: disable=too-many-arguments,too-many-locals
             )
 
 
-class DJWriter(DJReader):
+class DJBuilder(DJClient):
     """
-    Client class to manage all your DJ dag: nodes, attributes, namespaces, tags.
+    Client class for DJ node manipulation.
     """
+
+    #
+    # Catalogs and Engines
+    #
+    def list_catalogs(self):
+        """
+        Gets all catalogs.
+        """
+        response = self._session.get("/catalogs/", timeout=self._timeout)
+        return response.json()
+
+    def list_engines(self):
```

Comment:
Is listing engines on its own any value to client users? If we keep it, it might be better as an underscore method. Only because what users really care about is "what engine can I use for a specific catalog?" So they would list catalogs -> pull a catalog -> then look at the engines defined on it.

Reply:
I'd say it is useful because realistically most DJ server instances will be set up for one catalog, so this is like a convenience method where someone can say "give me all the engines for the primary catalog". Or maybe we can even have a "catalog" set when someone initializes the client, and just list all the engines from that catalog.